File: go/数组/1122. 数组的相对排序.go
package array

import "sort"

func relativeSortArray(arr1 []int, arr2 []int) []int {
	var d = make(map[int]int)
	var tmp []int
	var res []int
	sort.Ints(arr1)
	for _, v := range arr1 {
		if _, ok := d[v]; !ok {
			d[v] = 1
		} else {
			d[v]++
		}
		if inSlice(v, arr2) == false {
			tmp = append(tmp, v)
		}
	}
	for _, v := range arr2 {
		for i := 0; i < d[v]; i++ {
			res = append(res, v)
		}
	}
	res = append(res, tmp...)
	return res
}

func inSlice(e int, slice []int) bool {
	for _, s := range slice {
		if s == e {
			return true
		}
	}
	return false
}

File: python/数组/1122. 数组的相对排序.py
from typing import List
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        d = dict()
        for i in arr1:
            if i not in d:
                d[i] = 1
            else:
                d[i] += 1
        arr1.sort()
        tmp = []
        for i in arr1:
            if i not in arr2:
                tmp.append(i)
        res = []
        for i in arr2:
            res += [i]*d[i]
        return res + tmp

s = Solution()
arr1 = [2,3,1,3,2,4,6,7,9,2,19]
arr2 = [2,1,4,3,9,6]
print(s.relativeSortArray(arr1,arr2))


